website/content/docs/nia/network-drivers/terraform.mdx

```diff
@@ -0,0 +1,58 @@
+
+### Local Workspaces
+
+Within the Consul-Terraform-Sync configuration for a task, practitioners can select the desired module to run for the task as well as set the condition to execute the task. Each task executed by the Terraform driver corresponds to an automated root module that calls the selected module in an isolated Terraform environment. Consul-Terraform-Sync will manage concurrent execution of these tasks.
+
+Autogenerated root modules for tasks are mantained in local subdirectories of the Consul-Terraform-Sync working directory. Each subdirectory represents the local workspace for a task. By default, the working directory `sync-tasks` is created in the current directory. To configure where Terraform configuration files are stored, set [`working_dir`](/docs/nia/configuration#working_dir) to the desired path or configure the [`task.working_dir`](/docs/nia/configuration#working_dir-1) individually.
+
+~> **Note:** Although Terraform state files for task workspaces are independent, this does not guarantee the infrastructure changes from concurrent task executions are independent. Ensure that modules across all tasks are not modifying the same resource objects or have overlapping changes that may result in race conditions during automation.
+
+### Root Module
+
+The root module is simple in structure and proxies Consul information, configuration, and other variables to the Terraform module for the task. The content of the files that make up the root module are sourced from Consul-Terraform-Sync configuration, information for task's module to use as the automation playbook, and information from Consul such as service information.
+
+A working directory with one task named "cts-example" would have the folder structure below when running with the Terraform driver.
+
+```shell-session
+$ tree sync-tasks/
+
+sync-tasks/
+└── cts-example/
+    ├── main.tf
+    ├── variables.tf
+    ├── terraform.tfvars
+    └── terraform.tfvars.tmpl
+```
+
+The following files of the root module are generated for each task. An [example of a root module created by Consul-Terraform-Sync](https://github.com/hashicorp/consul-terraform-sync/tree/master/examples) can be found in the project repository.
+
+- `main.tf` - The main file contains the terraform block, provider blocks, and a module block calling the module configured for the task.
+  - `terraform` block - The corresponding provider source and versions for the task from the configuration files are placed into this block for the root module. The Terraform backend from the configuration is also templated here.
+  - `provider` blocks - The provider blocks generated in the root module resemble the `terraform_provider` blocks from the configuration for Consul-Terraform-Sync. They have identical arguments present and are set from the intermediate variable created per provider.
```

Comment:
Resemble. There's additional options that are available in `terraform_provider` and not `provider`.

Terraform config
```hcl
provider {
  // normal Terraform provider options
  address = ""
}
```

CTS config
```hcl
terraform_provider {
  // normal Terraform provider options
  address = ""

  // CTS only
  // load secrets from kv, vault, env to avoid writing values on disk
  task_env {
    "FOO_TOKEN" = "{{ key \"consulkvpath\" }}"
  }
}
```

Reply:
Thanks for clarifying. I was going to add a statement calling out the difference, but it might not provide enough informational value for the additional words. I guess we can update the docs if someone asks about the differences.